src/js/facility-locator/components/Select.jsx

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import Listbox from './Listbox';
+import {
+  getDirection,
+  getSelection,
+  isSelect,
+  isToggle,
+  isTraverse,
+  noServices
+} from '../utils/helpers.js';
+
+class Select extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      dropdownActive: false,
+      focusedIndex: 0
+    };
+    this.options = [];
+  }
+
+  handleFilterSelect = (newOption, type) => {
+    const { currentQuery } = this.props;
+    const { facilityType, serviceType } = currentQuery; // eslint-disable-line no-unused-vars
+    if (type === 'facility' && newOption !== facilityType) {
+      if (newOption === 'AllFacilities') {
+        this.props.updateSearchQuery({
+          facilityType: null,
+          serviceType: null
+        });
+      } else {
+        this.props.updateSearchQuery({
+          facilityType: newOption,
+          serviceType: null
+        });
+      }
+    }
+    if (type === 'service' && newOption !== serviceType) {
+      if (newOption === 'All') {
+        this.props.updateSearchQuery({
+          serviceType: null,
+        });
+      } else {
+        this.props.updateSearchQuery({
+          serviceType: newOption
+        });
+      }
+    }
+    return this.toggleDropdown(null, type);
+  }
+
+  navigateDropdown = (e) => {
+    const type = this.props.optionType;
+    const isActive = this.state.dropdownActive;
+    const which = e.target;
+    if (isToggle(e, isActive)) {
+      if (isActive && isSelect(e.keyCode)) {
+        this.handleFilterSelect(which.id, type);
+      }
+      return this.toggleDropdown(e, type);
+    }
+    if (isActive && isTraverse(e)) {
+      const increment = getDirection(e.keyCode);
+      const newIndex = this.state.focusedIndex + increment;
+      return this.focusOption(this.options[newIndex], newIndex, type);
+    }
+    return false;
+  }
+
+  handleBlur = (e) => {
+    const { currentTarget } = e;
+    setTimeout(() => {
+      if (!currentTarget.contains(document.activeElement)) {
+        this.setState({ dropdownActive: false });
+      }
+    });
+  }
+
+  toggleDropdown = (e) => {
+    const type = this.props.optionType;
+    const isActive = this.state.dropdownActive;
+    const { currentQuery: { facilityType, serviceType } } = this.props; // eslint-disable-line no-unused-vars
+    const queryType = type === 'service' ? serviceType : facilityType;
+    if (type === 'service' && noServices(type, facilityType)) {
```

Comment:
If we put a disabled check in the onClick handler, could this check go away?

Reply:
Yes good idea.